src/compas/geometry/primitives/frame.py

```diff
@@ -180,6 +180,8 @@ def __iter__(self):
         return iter([self.point, self.xaxis, self.yaxis])
 
     def __eq__(self, other, tol=1e-05):
+        if not hasattr(other, '__iter__') or not hasattr(other, '__len__') or len(self) != len(other):
+            return False
         for v1, v2 in zip(self, other):
             for a, b in zip(v1, v2):
                 if math.fabs(a - b) > tol:
```

Comment:
should we not replace this with the precision setting of `compas`?
noticed that we are not doing this for the other checks either...

Reply:
Yeah, that makes sense.

Reply:
actually, i'm a little confused by this comment.  isn't `compas.PRECISION` more about formatting the way objects are printed rather than prescribing tolerances on computations?

Reply:
yes and no. it is, for example, also used in the geometric maps to determine comparison precision...
but i agree that there is no "official policy" for this at the moment.

Reply:
but in the tests, when geometry comparisons are concerned, we tend to use (or should be using) `allclose` to avoid tests failing for lack of precision

Reply:
yeah, i've changed to using `allclose` now, but that also uses a hardcoded tolerance...

maybe once there is an "official policy" on this, such changes can be made more thoroughly.